From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sybren=20A=2E=20St=C3=BCvel?= <sybren@blender.org>
Date: Thu, 17 Dec 2020 13:23:59 +0100
Subject: [PATCH] Explicitly link X11 libraries

Fix X11 library underlinking, which was breaking Debian and Ubuntu
packages.

From Ubuntu Hirsute changelog:

```
blender (2.83.5+dfsg-4ubuntu1) hirsute; urgency=medium

  * Try to also link ghost library with x11, needed because of missing
    XConvertSelection symbol link (used in ghost static library).
  * Don't use gold, but switch to bfd linker that seems to be working better
    on ppc64el.

 -- Gianfranco Costamagna <locutusofborg@debian.org>  Wed, 11 Nov 2020 14:17:29 +0100
```

Reviewed by: sybren

Differential Revision: https://developer.blender.org/D9617
---
 intern/ghost/CMakeLists.txt | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/intern/ghost/CMakeLists.txt b/intern/ghost/CMakeLists.txt
index 77e777d..758b3c0 100644
--- a/intern/ghost/CMakeLists.txt
+++ b/intern/ghost/CMakeLists.txt
@@ -183,6 +183,11 @@ elseif(WITH_GHOST_X11 OR WITH_GHOST_WAYLAND)
       ${X11_X11_INCLUDE_PATH}
     )
 
+    list(APPEND LIB
+      ${X11_X11_LIB}
+      ${X11_Xrender_LIB}
+    )
+
     list(APPEND SRC
       intern/GHOST_DisplayManagerX11.cpp
       intern/GHOST_SystemX11.cpp
@@ -234,6 +239,9 @@ elseif(WITH_GHOST_X11 OR WITH_GHOST_WAYLAND)
       list(APPEND INC_SYS
         ${X11_xf86vmode_INCLUDE_PATH}
       )
+      list(APPEND LIB
+        ${X11_Xf86vmode_LIB}
+      )
     endif()
 
     if(WITH_X11_XFIXES)
@@ -241,6 +249,9 @@ elseif(WITH_GHOST_X11 OR WITH_GHOST_WAYLAND)
       list(APPEND INC_SYS
         ${X11_Xfixes_INCLUDE_PATH}
       )
+      list(APPEND LIB
+        ${X11_Xfixes_LIB}
+      )
     endif()
 
     if(WITH_X11_ALPHA)
@@ -252,6 +263,9 @@ elseif(WITH_GHOST_X11 OR WITH_GHOST_WAYLAND)
       list(APPEND INC_SYS
         ${X11_Xinput_INCLUDE_PATH}
       )
+      list(APPEND LIB
+        ${X11_Xinput_LIB}
+      )
     endif()
 
     add_definitions(-DWITH_GHOST_X11)
